extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -1202,4 +1202,24 @@ describes.realWin('amp-story', {
           });
     });
   });
+  describe('amp-story branching', () => {
+    it('should navigate to the target page when a goToPage action is executed',
+        () => {
+          createPages(story.element, 4,
+              ['cover', 'page-1', 'page-2', 'page-3']);
+          story.buildCallback();
+          return story.layoutCallback()
+              .then(() => {
+                const actionButton = createElementWithAttributes(
+                    win.document,
+                    'button',
+                    {'id': 'actionButton',
+                      'on': 'tap:cover.goToPage(id=page-2)'});
```

Comment:
Just a question about how publishers will have to use this: it'll have to be `tap:${pageId}.goToPage(...)`?
Is there a way to make it so they don't have to know the current `pageId` to write the action?

Reply:
Yes, this is correct!  Unfortunately, because we want this to be an amp action, [based on the actions and events documentation](https://www.ampproject.org/docs/interaction_dynamic/amp-actions-and-events), the targetId is required.

Reply:
Can't the ID be the ID of the button?

Reply:
Would it be feasible as a future change to implement the action on the `<amp-story>` element to avoid needing to reference the specific page?

Reply:
@newmuis I believe that the targetId should be the dom element that we'd like to execute an action _on_. The go to action really just switches the page that is currently active which is handled by the `amp-story-page` element (and then at the amp-story level). I think the action needs to be bound to a page element. Just to be sure, I did a quick test and it doesn't seem to work with the id of the button. 

@cvializ I think that would be possible, since the switching method is handled both in the story and page level. Though, I'm not sure how we feel about the fact that this would require giving the `amp-story` element an id (would users typically do this?), whereas a pageId is always required to make a story.

Reply:
Whoops, makes sense.  One data point here is that amp-carousel has a similar API ("go to slide") which uses the carousel ID.  I think semantically the story ID may make more sense ("the story should go to page 5" rather than "page 1 should go to page 5")